README.md

```diff
@@ -0,0 +1,142 @@
+# Pactum
+## The HTTP-API specification sketchbook for pythonistas
+
+[![Circle CI](https://circleci.com/gh/olist/pactum.svg?style=svg)](https://circleci.com/gh/olist/pactum)
+
+```shell
+pip install pactum
+```
+
+With `Pactum` you can **specify** HTTP-APIs using pure python.
+
+Pactum is easy to use, easy to extend and easy to contribute:
+
+### Easy to use
+
+The only requirements to start writing an API specification with `pactum`
+is `pactum` package itself and some knowledge of python.
+
+```python
+import pactum
+
+class MyAPI(API):
+    name = 'My API'
+    versions = [...]
+```
+
+### Easy to extend
+
+Using the [visitor pattern](http://wiki.c2.com/?VisitorPattern) you can create
+exporters and extensions for any format or service you want.
+
+Take a look at [pactum/exporters/openapi.py](https://github.com/olist/pactum/blob/master/pactum/exporters/openapi.py).
+
+### Architecture
+
+Always keep [this diagram](https://github.com/olist/pactum/wiki/Architecture-Diagram) in mind when defining your APIs.
+
+
+### Tutorial
+
+Create a file called specs.py and start defining your API.
+
+You can define a `Resource` object for your API.
+
+```python
+from pactum import Action, API, Resource, Response, Version
+from pactum import fields, verbs
+
+class OrderResource(Resource):
+    fields = [
+        fields.IntegerField(name='code', required=True),
+        fields.TimestampField(name='created_at'),
+        fields.StringField(name='item')
+    ]
+resource = OrderResource()
+
+error_resource = Resource(
+    name = 'ErrorResource'
+    fields = [fields.StringField(name='error', required=False)]
+)
+```
+You can define any element of your specification by calling it directly as in
+`error_resource` or by class definition as in `MyResource` and then calling it.
+
+
+List resources are definitions of lists of the same resource.
+```python
+list_order_resource = ListResource(resource=resource)
+```
+
+You can define `Response` objects with `status`, `description`(optional)  a
+`header`(optional) and a `Resource`/`ListResource` object as `body` (optionally)...
+
+```python
+ok_response = Response(
+    status=200, description='Here is your orders list.', body=list_resource
+)
+
+error_response = Response(status=400, resource=error_resource, headers=[('Content-type': 'application-json')])
+```
+
+... and `Request` objects with `verb`, `description`, `header`(optional) and a `Resource`/`ListResource`
+object as `payload`.
+
+```python
+request = Request(verb=verbs.GET, payload=resource)
+```
+
+An `Action` groups your request and a list of responses for an specified action
+passed in description parameter.
+```python
+action = Action(
+    description='Returns a list of resources.',
+    request=request,
+    responses=[error_response, ok_response]
```

Comment:
maybe it would be interesting to allow generic responses only by specifying the status code for simplicity's sake, e.g.

```python
responses=[200, 304, 400, 404, ...]
```

maybe eve

Reply:
We've some plans to make it easier with 2 new features:

1. REST-related helper class to be reused in a kind-of `contrib` package (remember that pactum must provide functionalities do spec all kind of APIs not only restful APIs)
2. A 'composite/container' inheritance mechanism that allow us to define global specs at a higher level of spec (like... define response codes at API level)

Reply:
I think that REST helpers can solve a lot of this problems.